File: zd1.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	fmt.Println("Введите текст")
	var text string
	fmt.Scanln(&text)

	for i := 0; i < len(text)-2; i++ {
		if text[i] == text[i+1] && text[i+1] == text[i+2] {
			text = text[:i+2] + string(randomChar()) + text[i+3:]
		}
	}

	fmt.Println(text)
}

func randomChar() rune {
	rand.Seed(time.Now().UnixNano())
	chars := []rune("abcdefghijklmnopqrstuvwxyz")
	return chars[rand.Intn(len(chars))]
}

File: zd1.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <ctime>
using namespace std;
   
namespace LAB2 {
    class StringValidator {
    public:
        static void Mai() {
            setlocale(LC_ALL, "Russian");
            cout << "Введите текст" << endl;
            string text;
            getline(cin, text);
                        //Он сканирует строку `text` и заменяет любую последовательность из трех одинаковых символов случайным символом.
            for (int i = 0; i < text.length() - 2; i++) {
                if (text[i] == text[i + 1] && text[i + 1] == text[i + 2]) {
                //    text.erase(i + 2, 1);
                    text.insert(i + 2, 1, RandomChar()); 
                }
            }

            cout << text << endl;
        }

    private:
        static char RandomChar() {
            static const char chars[] = "abcdefghijklmnopqrstuvwxyz";
            int charIdx = rand() % 26;
            return chars[charIdx];
        }
    };
}

int main() {
    srand(static_cast<unsigned>(time(nullptr)));
    LAB2::StringValidator::Mai();

    return 0;
}
